src/plugins/data/public/query/saved_query/saved_query_service.ts

```diff
@@ -119,12 +119,15 @@ export const createSavedQueryService = (
     id: string;
     attributes: SerializedSavedQueryAttributes;
   }) => {
-    let queryString;
+    let queryString = savedQuery.attributes.query.query;
+
     try {
-      queryString = JSON.parse(savedQuery.attributes.query.query);
-    } catch (error) {
-      queryString = savedQuery.attributes.query.query;
-    }
+      const parsedQueryString = JSON.parse(savedQuery.attributes.query.query);
+      if (typeof parsedQueryString === 'object') {
```

Comment:
This check will pass if we have `'null'` in `savedQuery.attributes.query.query`. Is that okay?

Reply:
@dokmic great catch!
Fixed